File: docs/src/test/java/jdocs/home/persistence/HelloEvent.java
/*
 * Copyright (C) 2009-2018 Lightbend Inc. <https://www.lightbend.com>
 */
package jdocs.home.persistence;

import com.lightbend.lagom.javadsl.persistence.AggregateEvent;
import com.lightbend.lagom.javadsl.persistence.AggregateEventShards;
import com.lightbend.lagom.javadsl.persistence.AggregateEventTag;

public interface HelloEvent extends AggregateEvent<HelloEvent> {

  int NUM_SHARDS = 20;

  AggregateEventShards<HelloEvent> TAG = AggregateEventTag.sharded(HelloEvent.class, NUM_SHARDS);

  @Override
  default AggregateEventShards<HelloEvent> aggregateTag() {
    return TAG;
  }

  class GreetingMessageChanged implements HelloEvent {
    public final String name;
    public final String message;

    public GreetingMessageChanged(String name, String message) {
      this.name = name;
      this.message = message;
    }
  }
}

File: docs/src/test/scala/docs/home/persistence/HelloEvent.scala
/*
 * Copyright (C) 2009-2018 Lightbend Inc. <https://www.lightbend.com>
 */
package docs.home.persistence
import com.lightbend.lagom.scaladsl.persistence.{AggregateEvent, AggregateEventShards, AggregateEventTag}

object HelloEvent {
  val NumShards = 20
  val Tag = AggregateEventTag.sharded[HelloEvent](NumShards)

  final case class GreetingChanged(name: String, message: String) extends HelloEvent
}

sealed trait HelloEvent extends AggregateEvent[HelloEvent] {
  override def aggregateTag: AggregateEventShards[HelloEvent] = HelloEvent.Tag
}
